Source/WebCore/platform/network/BlobResourceHandle.cpp

```diff
@@ -277,15 +307,15 @@ void BlobResourceHandle::didGetSize(long long size)
 
     // The size passed back is the size of the whole file. If the underlying item is a sliced file, we need to use the slice length.
     const BlobDataItem& item = m_blobData->items().at(m_sizeItemCount);
-    size = item.length();
+    size_t updatedSize = static_cast<size_t>(item.length());
```

Comment:
I don't fully understand why this function now completely ignores its `size` argument. Should we remove the argument if it's not needed?

I'm also unsure why the cast to `size_t` is both needed and guaranteed to work.

Reply:
Regarding the change in the casting, I agree that you made it no worse since we’ll eventually get to `size_t`. But I still think we should think about whether there’s a 32-bit bug here due to casting `long long` to `size_t` without any checking.

I didn’t see your thought on my other comment:
> Should we remove the argument if it's not needed?